osf_tests/test_queued_mail.py

```diff
@@ -149,3 +152,41 @@ def test_user_is_not_active_is_not_confirmed(self, mock_mail):
             mail=NO_ADDON,
         )
         assert mail.send_mail() is False
+
+    @mock.patch('osf.models.queued_mail.send_mail')
+    def test_remind_prereg_presend(self, mock_mail, user):
+        prereg = DraftRegistrationFactory(registration_schema=MetaSchema.objects.get(name='Prereg Challenge'))
+        mail = self.queue_mail(mail=PREREG_REMINDER, user=user, draft_id=prereg._id)
+        assert mail.send_mail()
+
+    @mock.patch('osf.models.queued_mail.send_mail')
+    def test_remind_prereg_presend_with_approval(self, mock_mail, user):
+        prereg = DraftRegistrationFactory(registration_schema=MetaSchema.objects.get(name='Prereg Challenge'))
+
+        approval = DraftRegistrationApproval(
+            meta={
+                'registration_choice': 'immediate'
+            }
+        )
+        approval.save()
+        prereg.approval = approval
+        prereg.save()
+
+        mail = self.queue_mail(mail=PREREG_REMINDER, user=user, draft_id=prereg._id)
+        assert not mail.send_mail()
+
+    @mock.patch('osf.models.queued_mail.send_mail')
+    def test_remind_prereg_presend_deleted_draft(self, mock_mail, user):
+        prereg = DraftRegistrationFactory(registration_schema=MetaSchema.objects.get(name='Prereg Challenge'))
+        mail = self.queue_mail(mail=PREREG_REMINDER, user=user, draft_id=prereg._id)
+        prereg.delete()
+        assert not mail.send_mail()
+
+    @mock.patch('osf.models.queued_mail.send_mail')
+    def test_remind_prereg_presend_already_sent(self, mock_mail, user):
+        prereg = DraftRegistrationFactory(registration_schema=MetaSchema.objects.get(name='Prereg Challenge'))
+        mail = self.queue_mail(mail=PREREG_REMINDER, user=user, draft_id=prereg._id)
+        mail.send_mail()
```

Comment:
If I'm reading this correctly, it looks like this test repeats all the steps of test 1-- this second half could probably be tacked onto the first test with a note saying `# test remind prereg presend already sent` to save some time.